### PR TITLE
refactor: establish target.Reconciler

### PR DESCRIPTION
**Pull request by erikgb**
This is a major refactor of the bundle target area. The main motivation is to prepare the grounds for https://github.com/cert-manager/trust-manager/issues/58 and https://github.com/cert-manager/trust-manager/issues/222.

~~The refactoring became larger because I had to move some fields and functions to avoid circular package dependencies. I would be happy to split these changes out in pre-PRs if desired.~~ Multiple pre-PRs merged now.

### CODE REVIEW

File: pkg/bundle/internal/target/target.go [hunk 1]
```diff
@@ -354,31 +367,31 @@ func prepareTargetPatch[T targetApplyConfiguration[T]](target T, bundle trustapi
 		)
 }
 
-type targetData struct {
-	data       string
-	binaryData map[string][]byte
+type Data struct {
+	Data       string
+	BinaryData map[string][]byte
 }
```
Comment (SgtCoDFish) [inline pkg/bundle/internal/target/target.go:373]: comment: Looking at this now, I think we could use better names. Obviously it's not related to this PR, but `Data.PEM` and `Data.Binary` or something would be more descriptive and avoid echoing `Data.Data` 🤔

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

This looks a good bit cleaner to me. I like the API boundary you've established with target.Reconciler!

Thanks for this 😁

File: pkg/bundle/internal/target/target.go (hunk 1 above)
Comment (erikgb) [reply pkg/bundle/internal/target/target.go:373]: Agreed! I can try to fix this in one of my follow-up PRs.

Comment (inteon) [reply pkg/bundle/internal/target/target.go:373]: Ah, the data struct is unlike the non-data struct. 😄

**Pull request merged**